code/modules/hydroponics/seeds.dm

```diff
@@ -112,56 +112,48 @@
 	if(prob(traitmut))
 		add_random_traits(1, 1)
 
-
-
-/obj/item/seeds/bullet_act(obj/item/projectile/Proj) //Works with the Somatoray to modify plant variables.
+/obj/item/seeds/bullet_act(obj/item/projectile/Proj) // Works with the Somatoray to modify plant variables.
 	if(istype(Proj, /obj/item/projectile/energy/florayield))
 		var/rating = 1
 		if(istype(loc, /obj/machinery/hydroponics))
 			var/obj/machinery/hydroponics/H = loc
 			rating = H.rating
 
-		if(yield == 0)//Oh god don't divide by zero you'll doom us all.
+		if(yield == 0) // Oh god don't divide by zero you'll doom us all.
 			adjust_yield(1 * rating)
-		else if(prob(1/(yield * yield) * 100))//This formula gives you diminishing returns based on yield. 100% with 1 yield, decreasing to 25%, 11%, 6, 4, 2...
+		else if(prob(1/(yield * yield) * 100)) // This formula gives you diminishing returns based on yield. 100% with 1 yield, decreasing to 25%, 11%, 6, 4, 2...
 			adjust_yield(1 * rating)
 	else
 		return ..()
 
-
 // Harvest procs
 /obj/item/seeds/proc/getYield()
 	var/return_yield = yield
 
 	var/obj/machinery/hydroponics/parent = loc
 	if(istype(loc, /obj/machinery/hydroponics))
 		if(parent.yieldmod == 0)
-			return_yield = min(return_yield, 1)//1 if above zero, 0 otherwise
+			return_yield = min(return_yield, 1) // 1 if above zero, 0 otherwise
 		else
 			return_yield *= (parent.yieldmod)
 
-	return return_yield
-
+	return CEILING(return_yield, 1) // No decimal plants, please
 
 /obj/item/seeds/proc/harvest(mob/user = usr)
-	var/obj/machinery/hydroponics/parent = loc //for ease of access
-	var/t_amount = 0
-	var/list/result = list()
```

Comment:
is there a reason why you're getting rid of the list that was returned at the end of the proc?